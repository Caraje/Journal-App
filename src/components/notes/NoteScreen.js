import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { activeNote, startDeleting } from '../../actions/notes'
import { useForm } from '../../hooks/useForm'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
	const {notes: {active:note }} = useSelector( state => state )
	const [ formValues, handleInputChange, reset ] = useForm( note )
	const { body, title, id } = formValues

	const activeId = useRef( note.id )
	const dispatch = useDispatch()

	useEffect( () => {
		if( note.id !== activeId.current ) {
			reset( note )
			activeId.current = note.id
		}
	}, [ note, reset ] )

	useEffect( () => {
		dispatch( activeNote( formValues.id, {...formValues } ) )
	}, [ formValues, dispatch ] )
    
	const handleDelete = () => {

		dispatch( startDeleting( id ) )
	}

	return (
		<div className='notes__main-content'>
			<NotesAppBar />
			<div className='notes__content'>

				<input
					type='text'
					placeholder='Some Awesome title'
					className='notes__tite-input'
					name='title'
					autoComplete='off'
					value={ title}
					onChange={handleInputChange}
				/>

				<textarea
					placeholder='What hapened Today'
					className='note__textarea'
					name='body'
					autoComplete='off'
					value={ body}
					onChange={handleInputChange}
				>   

				</textarea>

				{
					( note.url ) &&
                            ( 
                            	<div className='note__image mt-5'>
                            		<img
                            			src={`${ note.url }`}
                            			alt='imagen'
                            		/>

                            	</div>
                            )
				}
                

			</div>

			<button
				className='btn btn-danger'
				onClick={ handleDelete }
			>
                Delete
			</button>
		</div>
	)
}
