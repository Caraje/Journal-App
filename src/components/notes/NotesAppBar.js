import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startSaveNote, startUploading } from '../../actions/notes'
import moment from 'moment'


export const NotesAppBar = () => {
	const dispatch = useDispatch()
	const {notes:{ active }} = useSelector( state => state )
	const noteDate = moment( active.date )

	const handleSave = () => {
		dispatch( startSaveNote( active ) )
	}

	const handlepictureClick = () => {

		document.querySelector( '#fileSelector' ).click()

	}

	const handleFileChange = ( e ) => {
		const file = e.target.files[0]

		if ( file ) {
			dispatch( startUploading( file ) )
		}
		// console.log(file)
	}


	return (
		<div className='notes__appbar'>
			<span>{noteDate.format( 'D MMMM YYYY ' )} </span>
			<input
				id='fileSelector'
				type='file'
				name='file'
				style= {{ display: 'none'}}
				onChange={ handleFileChange }
			/>
			<div>
				<button 
					className='btn'
					onClick={ handlepictureClick}
				>
                Picture
				</button>
				<button
					className='btn'
					onClick={ handleSave }
				>
                Save
				</button>
			</div>
		</div>
	)
}
