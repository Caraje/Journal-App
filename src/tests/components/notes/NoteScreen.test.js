import { mount } from 'enzyme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk'
import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/notes/NoteScreen';

jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn()
}))


const middlewares = [ thunk ]
const mockStore = configureStore( middlewares )
const initState = {
    auth: {},
    ui: {
        loading: false, 
        msgError: null
    },
    notes: {
        active: {
            id: 'ABC',
            title: 'Hola',
            body: 'Mundo',
            date: 0
        },
        notes: []
    }
}
let store = mockStore( initState )
store.dispatch = jest.fn()


const wrapper = mount(
    <Provider store = { store }>
        <NoteScreen />
    </Provider>
) 

describe('Pruebas en NotesScreen', () => {
    
    test('debe mostrar correctamente el componente', () => {
        expect( wrapper ).toMatchSnapshot()
    });

    test('debe disparar el activeNote', () => {
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title', 
                value: 'Hola de nuevo'
            }
        })

        expect( activeNote ).toHaveBeenLastCalledWith(
            'ABC',
            {
                body: 'Mundo',
                title: 'Hola de nuevo',
                id: 'ABC',
                date: 0
            }
        )
    });
});