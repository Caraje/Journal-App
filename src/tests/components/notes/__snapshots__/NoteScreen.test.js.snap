// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en NotesScreen debe mostrar correctamente el componente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      1 January 1970 
       
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__tite-input"
      name="title"
      onChange={[Function]}
      placeholder="Some Awesome title"
      type="text"
      value="Hola"
    />
    <textarea
      autoComplete="off"
      className="note__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What hapened Today"
      value="Mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
